### PR TITLE
update `markup_text_boxst`

### PR DESCRIPTION
**Pull request by ab9rf**
this PR is based on bay12 header files and addresses issues raised in #569 and #574

### CODE REVIEW

**Comment by ab9rf** [comment]
crap wrong base branch, sorry

**Comment by Bumber64** [comment]
Should we maintain the init-values for `markup_text_wordst`? Those are set in a constructor function. Same for `markup_text_linkst`'s `id` and `subid`, unless those default to -1 automatically.

Is it possible to do a refers-to on `link_index` such that gui/gm-editor can jump to the ref, or does that only work when something's global?

**Comment by ab9rf** [comment]
i'll add the init-values, i missed them working from toady's source because he uses a constructor function to set them instead of specifying them as defaults (as a note: you are free to use github's code review mechanism to suggest changes to a pending PR, that's why we use PRs)

the problem with making `link-index` jumpable in `gm-editor` is that the logic to figure out what to jump to is probably more complicated than `gm-editor` is currently capable of understanding

File: df.ui-menus.xml [hunk 1]
```diff
@@ -2663,30 +2663,60 @@
...
+        <enum-item name='ART_IMAGE'/>
+        <enum-item name='ERA'/>
+        <enum-item name='HEC'/>
+    </enum-type>
+
+    <struct-type type-name='markup_text_linkst'>
+        read and write serialization is in order as below
+        <enum type-name='markup_text_link_type' name='type'/>
+        <int32_t name='id' init-value='-1'/>
+        <int32_t name='subid' init-value='-1'/>
```
Comment (Bumber64) [inline df.ui-menus.xml:2703]: **Comments:**
subid: only used for type ABSTRACT_BUILDING and ART_IMAGE

File: df.ui-menus.xml [hunk 2]
```diff
@@ -2663,30 +2663,60 @@
...
 
+    <struct-type type-name='markup_text_wordst'>
+        read and write serialization is in order as below
+        <stl-string name='str'/>
+        <uint8_t name='red' init-value='255'/>
+        <uint8_t name='green' init-value='255'/>
+        <uint8_t name='blue' init-value='255'/>
+        <int32_t name='link_index' init-value='-1'/>
+        <int32_t name='px' init-value='-1'/>
+        <int32_t name='py' init-value='-1'/>
```
Comment (Bumber64) [inline df.ui-menus.xml:2674]: **Comments:**
link_index: refers to this text box's link vector, makes clickable link in legends mode
px: offset from left of text box in characters
py: offset from top of text box (positive is downwards)

**Review by Bumber64 — COMMENTED** [review]
Some comments on the purpose of some fields.

**Comment by Bumber64** [comment]
You left in the "subid: " part of the subid comment and left out the ending ')' for py.

**Review by myk002 — APPROVED** [review]

**Pull request merged**